### ISSUE TITLE
[Bug report] navigate to anchor on another page does not work

### ISSUE THREAD

**Issue by John60676** [vuepress/vuepress-next#77]
### Description

Navigation to anchor on another page does not work. In dev mode, the browser will receive a warning message from vue-router. 

```
  Couldn't find element using selector "#plugins" returned by scrollBehavior.
```

### Steps to reproduce

- Reproduction link / repo: https://vuepress2.netlify.app/guide/plugin.html

### Expected behavior

Navigation to anchor 

### Screenshots

![vuepress-next](https://user-images.githubusercontent.com/21698973/112244382-2d668100-8c8a-11eb-9b6c-016b8bee397e.gif)

### Environment info

- Browser: Chrome 89.0.4389.90

**Comment by meteorlxy** [comment]
In fact I have noticed it before. Just didn't have time to figure out the reason. 😅

**Comment by meteorlxy** [comment]
Not sure if this is related to vue-router, cc @posva

**Comment by posva** [comment]
This is because of the transition. When using a transition, you need to return a promise in `scrollBehavior` at https://github.com/vuepress/vuepress-next/blob/187aef36607efc62d7b2d5c773553f89685cf64c/packages/%40vuepress/client/src/app.ts#L78

There is an example at https://github.com/vuejs/vue-router-next/tree/master/e2e/scroll-behavior by using `@before-enter` and `@before-leave` on the transition wrapping the view

**Comment by meteorlxy** [comment]
@posva Thanks for your tips.

Transitions may be added by themes, so the `scrollBehavior` should be modified on demand by theme. So https://github.com/vuejs/vue-router-next/pull/602 is really helpful lol 👍

**Comment by John60676** [comment]
Maybe I can submit a PR to fix this problem

**Comment by meteorlxy** [comment]
@posva 

We found that the initial scrollBehavior will also fail, because it happens before the app mounted. Any ideas about that?

![image](https://user-images.githubusercontent.com/18205362/112570525-2a51c900-8e21-11eb-9b03-fd8660ccbb42.png)

![image](https://user-images.githubusercontent.com/18205362/112570586-42c1e380-8e21-11eb-856b-5276fdbaa3fd.png)

![image](https://user-images.githubusercontent.com/18205362/112570659-6edd6480-8e21-11eb-97ae-b09b97390083.png)

**Comment by posva** [comment]
> We found that the initial scrollBehavior will also fail, because it happens before the app mounted. Any ideas about that?

I checked on my e2e tests (scroll-behavior) and it seems to work correctly. Note you can check for the first navigation by checking if `from == START_LOCATION`

**Comment by meteorlxy** [comment]
The initial `scrollBehavior` issue is fixed via 2f5450f0b8dcc4aa49b1c19a1adea6e84a1594c4
